### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by myelin0**
## In this branch:

- We set up React Testing Library
- Write unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all React components.
- Run full tests suite using Jest. All tests should pass.
- Pass Linters.

### CODE REVIEW

File: src/logic/solve.test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  it('multiplication works properly', () => {
+    obj = {
+      total: '4',
+      next: '2',
+      operation: 'x',
+    };
+    const { total } = calculate(obj, '=');
+    expect(total).toBe('8');
+  });
+});
```
Comment (Emmyn5600) [inline src/logic/solve.test.js]: - [ ]  Please implement the following tests too, to cover all the features added in the project:

- It should give the percentage of the given number
- It should not calculate if the second operand doesn't exist.
- It should change the sign if clicked on a different sign.
- It should clear the display screen if clicked on the AC button.

File: src/logic/operation.test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  });
+  test('subtraction', () => {
+    const result = operate('10', '5', '-');
+    expect(result).toBe('5');
+  });
+  test('multiplication', () => {
+    const result = operate('2', '2', 'x');
+    expect(result).toBe('4');
+  });
+});
```
Comment (Emmyn5600) [inline src/logic/operation.test.js]: - [ ] Please implement the following tests too, to cover all the features added in the project:

 - It should throw 0 errors if trying to make a zero division.
 - It should return 20. as next if the button name is.
 - It should give the percentage of the first operand if the button name is %.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @myelin0  and @Christianib003 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
- Check the comment under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @myelin0, @Christianib003 👋🏻, 

Good job making the requested changes 💪🏻!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

✅ All tests pass 👍🏻 

Cheers and Happy coding!👏👏👏

you-did-it-good-job


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**